sys/net/gnrc/netif/ethernet/gnrc_netif_ethernet.c

```diff
@@ -162,7 +162,10 @@ static int _send(gnrc_netif_t *netif, gnrc_pktsnip_t *pkt)
 #endif
     res = dev->driver->send(dev, &iolist);
 
-    gnrc_pktbuf_release(pkt);
+    if (gnrc_netif_netdev_legacy_api(netif)) {
+        /* only for legacy drivers we need to release pkt here */
+        gnrc_pktbuf_release(pkt);
```

Comment:
`gnrc_netif_ieee802154.c` and `gnrc_netif_lorawan.c` probably also need an update

Reply:
Yes, indeed. It could be done either in this PR, or when the first drivers is converted. What do you prefer?

Reply:
If it's just a small change, just do it here so it's not missed in the future.

Reply:
I adapted it now for netif_raw, netif_ethernet, and netif_ieee802154. But for the LoRaWAN driver I do not dare to change. It doesn't free the buffer on success anyway. I am not sure if there is a leak, or if this behaves different for a reason.